### PR TITLE
feat: create UI components button

### LINKED ISSUE #12: Create UI components `Button`
## Description

- The task is to create a shared UI component for `button`.

## Criteria

- [ ] create `button` component 

## Notes

- https://ui.shadcn.com/docs/components/button
image

## Assignment 

|Role|Assignee |
|---|---|
|`Developer`|@KaiKoide |
|`Reviewer`|@Akikaze1119 |

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

Add button component in the components/ui folder based on the design on Figma.

## Changes

- Install button component from shadcn/ui(https://ui.shadcn.com/docs/components/button) 
- style the design

## Review points

- The styling matches the Figma design

## Screenshots or videos

The images of icons in Figma
image

button components
image

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/button.tsx [hunk 1]
```diff
@@ -5,24 +5,19 @@ import { cva, type VariantProps } from 'class-variance-authority';
...
-        destructive:
-          'bg-destructive text-destructive-foreground hover:opacity-80',
-        outline:
-          'border border-input bg-background hover:bg-accent hover:text-accent-foreground',
-        secondary: 'bg-secondary text-secondary-foreground hover:opacity-80',
-        ghost: 'hover:bg-accent hover:text-accent-foreground',
-        link: 'text-primary underline-offset-4 hover:underline',
+        default: 'bg-primary text-white hover:opacity-80',
+        secondary:
+          'border border-primary bg-white hover:bg-slate-300 hover:opacity-50 hover:border-none ',
```
Comment (Akikaze1119) [inline src/components/ui/button.tsx]: Could you please confirm that the border should be set as follows when the button is hovered.

Screenshot 2024-03-27 at 2 16 45 PM

File: src/components/ui/button.tsx [hunk 2]
```diff
@@ -5,24 +5,19 @@ import { cva, type VariantProps } from 'class-variance-authority';
...
+        disable: 'bg-white border border-color-gray-500 text-gray-500',
+        destructive: 'bg-rose-600 text-white hover:opacity-80',
```
Comment (Akikaze1119) [inline src/components/ui/button.tsx:16]: The Text Button component needs to be added as well, can you please add here?
I think you can use [Link component](https://ui.shadcn.com/docs/components/button#link-1)

▼Figma
Screenshot 2024-03-27 at 2 20 56 PM

**Review by Akikaze1119 — CHANGES_REQUESTED** [review]
Thank you for your work!
I added 2 comments, I would like you to check them.

**Note: About black color**
After the modified CSS file merge, it is recognized to be #333.

File: src/components/ui/button.tsx (hunk 1 above)
Comment (KaiKoide) [reply src/components/ui/button.tsx]: Thank you for your review!
I added the border when hovered.

File: src/components/ui/button.tsx (hunk 2 above)
Comment (KaiKoide) [reply src/components/ui/button.tsx:16]: I added the text button!

Comment (Akikaze1119) [reply src/components/ui/button.tsx:16]: @KaiKoide 
Thank you!
The font-size for link component is 16px, could you modify it?
(I think default font-size is 18px, so it's needed to add style)
https://capture.dropbox.com/bq15meM0GN8JwFta

**Review by Akikaze1119 — CHANGES_REQUESTED** [review]

**Review by Akikaze1119 — APPROVED** [review]
Thank you for modifying!
LGTM :)

**Pull request merged**